pkg/controller/controller_test.go

```diff
@@ -3646,3 +3648,23 @@ func addGetSecretReaction(fakeKubeClient *clientgofake.Clientset, secret *corev1
 		return true, secret, nil
 	})
 }
+
+var detector cache.CacheMutationDetector
+var locker sync.Mutex
+
+func getMutationDetector() cache.CacheMutationDetector {
+	if detector == nil {
+		locker.Lock()
+		defer locker.Unlock()
+		if detector != nil {
+			return detector
+		}
+		detector = cache.NewCacheMutationDetector("controller")
+		ch := make(<-chan struct{})
+		go func() {
+			detector.Run(ch)
+		}()
```

Comment:
this leaks goroutines. Channel never closed.

Reply:
yes, channel not closed. I will checki t.